### PR TITLE
Add nil guards for KV deserialization

### PR DESCRIPTION
**Pull request by peakxu**
@joshchu, @mguymon

### CODE REVIEW

File: lib/streamingly/serde.rb [hunk 1]
```diff
@@ -34,6 +34,7 @@ def self.from_csv(string)
 
     def self.from_tabbed_csv(string)
       k, v = string.split("\t", 2)
+      return if k.nil? || v.nil?
       key = from_string_or_csv(k)
```
Comment (mguymon) [inline lib/streamingly/serde.rb]: So the design constraint is a tabbed csv must always define the key and val, e.g.

    FancyKey,an-id\tFancyData,1,2,3

and the following is invalid

    FancyKey,an-id\t

**Comment by joshchu** [comment]
image

File: lib/streamingly/serde.rb (hunk 1 above)
Comment (peakxu) [inline lib/streamingly/serde.rb]: Before we implemented the batch view path method in Hadooply, I certainly saw quite a few cases of empty keys being written out as we only wanted one part. This should be happening less, but in any case, I don't think this SerDe should be exploding in this manner. Alternately, we may want to throw custom exception to this effect that the invariant is violated and then handle it in the generic mapper or reducer. Perhaps that's better? We can badger on the invalid input etc.

Comment (mguymon) [inline lib/streamingly/serde.rb]: I like the idea of raising an exception rather than silently failing

**Comment by mguymon** [comment]
+1

We should review streamingly and see if there are ways to propagate "issues" to the caller.

**Pull request merged**